inbm-v5.md

```diff
@@ -0,0 +1,320 @@
+
+1. #### inbc
+
+   - **Function**: In-band manageability's commandline interface
+   - **Main Tasks**:
+      - `inbc` acts as the commandline interace to other `previlaged` user-space applications to perform device-management actions (like os updates or firmware update etc) on the underlying host.
+      - a `trusted client` application which communicates with `inbm-daemon` over unix-sockets, translating manageability commands into gRPC API calls.
+   - **Example Use**:
+
+      ```code
+        inbc sota {--uri, -u=URI} 
+        [--releasedate, -r RELEASE_DATE; default="2026-12-31"] 
+        [--username, -un USERNAME]
+        [--mode, -m MODE; default="full", choices=["full","no-download", "download-only"] ]
+        [--reboot, -rb; default=yes]
+        [--package-list, -p=PACKAGES]
+      ```
+
+    For detailed usage of `inbc` refer to ![inbc usage guide](inbc-readme-link)
+
+1. #### cloud-client
+
+   - **Function**: Cloud `device management service` (DMS) connecting thread
+   - **Main Tasks**:
+      - North-bound acts MQTT client connecting to DMS (e.g. Azure IOT Central or ThingsBoard)
+      - South-bound acts as `inbm-daemon` client translating over-the-air (ota) commands from DMS to `inbm-daemon` gRPC API's
+      - Checks on any `state` file to perform additional tasks on startup, e.g. post a OS update related bootup.
+
+1. #### dispatcher-queue
+
+   - **Function**: Management command queue
+   - **Main Tasks**:
+      - implements a simple queue of size `1` for device management commands
+      - invokes `updater` thread based on the type of update command e.g. firmware or os or application
+
+1. #### updater threads
+
+   - **Function**: A `transieant` thread performing update on underlying host
+   - **Main Tasks**:
+      - _Firmware updater_: Perfomrs firmware update related tasks like:
+         - check applicability, i.e. verdor, version and date checks
+         - download capsule file and perform signature checks if applicable
+         - invoke IBV's firmware update tool based on firmware-update config file look up.
+         - update logging and state files
+         - send intermediate results to `inbm-daemon` for reporting
+         - trigger reboot of platform if applicable
+      - _OS updater_: Perfomrs OS update related tasks like:
+         - check applicability, e.g. checks available disk space
+         - download OS image file and perform signature checks if applicable
+         - invoke OS update tool based on underlying OS type/distribution.
+         - update logging and state files
+         - send intermediate results to `inbm-daemon` for reporting
+         - trigger reboot of platform if applicable
+      - _Application updater_: Perfomrs application update related tasks like:
+         - check applicability, e.g. checks available disk space
```

Comment:
Are we still going to have the docker bench security checks or are we dropping that for this version?

Reply:
no right now. But maybe in phases. Our first priority is:
- OS update (Ubuntu/TiberOS)
- Application update (Ubuntu)
- INBC

Basically aligning to the requirements of INBM from ITEP